Issue fixed for bigint, numeric and decimal data types

deepakshi-mittal opened the pull request:
Task: BABEL-4315

### Description
 
Added different functions to handle int, smallest and bigint input 
Added test cases covering join, all data types, temp tables 
Changed error message for unsupported datatypes(ie non-integer types) 
Numeric and decimal to be treated as biting, same as that of create table behavior with identity column

This issue is related to BABEL-539

### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -**


* **Boundary conditions -**


* **Arbitrary inputs -**


* **Negative test cases -**


* **Minor version upgrade tests -**


* **Major version upgrade tests -**


* **Performance tests -**


* **Tooling impact -**


* **Client tests -**



### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

deepakshi-mittal committed:
Issue fixed for bigint, numeric, decimal
 
Added different functions to handle int, smallest and bigint input
Added test cases covering join, all data types, temp tables
Changed error message for unsupported datatypes(ie non-integer types)
Numeric and decimal to be treated as biting, same as that of create
table behavior with identity column

Task: BABEL-4315

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

forestkeeper commented on contrib/babelfishpg_tsql/src/pl_handler.c:
```diff
@@ -5070,6 +5071,39 @@ static char *get_oid_type_string(int type_oid){
+					con = (Const *)n;
+					val = (int64) DatumGetInt64(con->constvalue);
+					break;
+				}
```
add an else in here and throw out some error

deepakshi-mittal replied:
okay

forestkeeper commented on contrib/babelfishpg_tsql/src/pl_handler.c:
```diff
@@ -5094,19 +5128,19 @@ static List *transformSelectIntoStmt(CreateTableAsStmt *stmt, const char *queryS
 			TargetEntry *tle = (TargetEntry *)lfirst(elements);
 
-			if (tle->expr && IsA(tle->expr, FuncExpr) && strcasecmp(get_func_name(((FuncExpr *)(tle->expr))->funcid), "identity_into") == 0)
+			if (tle->expr && IsA(tle->expr, FuncExpr) && strncmp(get_func_name(((FuncExpr *)(tle->expr))->funcid), "identity_into", strlen( "identity_into")) == 0)
```
Would you mind add another test case using IDENTITY in upper case ?

forestkeeper commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -1898,14 +1901,32 @@ bbf_set_context_info(PG_FUNCTION_ARGS)
+/**
+ * Common function to get next value of sequence for Identity function in Select Into
+*/
+static int64 get_identity_next_value(void){
```
the '{' should be in the next line

basasairohan committed:
Fix output param values in reordered parameter list in which the parameter names are in mixed case (#1664)

Currently, if the argument names in the procedure are in mixed case, the OUTPUT params are not being set properly. This changes fixes the same.

Issues Resolved
BABEL-4235

Signed-off-by: Sai Rohan Basa bsrohan@amazon.com

deepakshi-mittal commented:
Closing this Pr because of Merge issues
